src/pages/index.js

```diff
@@ -0,0 +1,349 @@
+import whitexSrc from "../images/white-x.svg";
+
+// initialize the api
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "c13b185c-8fbc-40ad-bb6b-9e8300eb8355",
+    "Content-Type": "application/json"
+  }
+});
+
+// modals
+const addCardModal = document.querySelector("#add-card-modal");
+const cardSubmitButton = addCardModal.querySelector(".modal__submit-button");
+const addCardElement = addCardModal.querySelector(".modal__form");
+const addCardModalCloseButton = addCardModal.querySelector(
+  ".modal__close-button"
+);
+
+const cardTemplate = document.querySelector("#card-template");
+const cardNameInput = addCardModal.querySelector("#add-card-name-input");
+const cardLinkInput = addCardModal.querySelector("#add-card-link-input");
+const cardsList = document.querySelector(".cards__list");
+
+let selectedCard, selectedCardId;
+
+// Profile elements
+const avatar = document.querySelector('.profile__avatar');
+const profileEditButton = document.querySelector(".profile__edit-btn");
+const cardModalButton = document.querySelector(".profile__add-btn");
+const avatarModalButton = document.querySelector(".profile__avatar-btn");
+const profileNameElement = document.querySelector(".profile__name");
+const profileDescriptionElement = document.querySelector(
+  ".profile__description"
+);
+
+// Form elements
+const editProfileModal = document.querySelector("#edit-profile-modal");
+const editFormElement = editProfileModal.querySelector(".modal__form");
+const editProfileCloseButton = editProfileModal.querySelector(
+  ".modal__close-button"
+);
+const editModalNameInput = editProfileModal.querySelector(
+  "#profile-name-input"
+);
+const editModalDescriptionInput = editProfileModal.querySelector(
+  "#profile-description-input"
+);
+
+const deleteForm = document.querySelector("#delete-form");
+
+// delete form elements
+const deleteModal = document.querySelector("#delete-modal");
+const closeDeleteModalButton = document.querySelector("#close-modal-delete-button");
+const cancelDeleteButton = deleteForm.querySelector(".modal__submit-btn");
+
+// declare images and assign them to the right places
+const pencil = document.querySelector('.profile__edit-btn-logo');
+pencil.src = pencilSrc;
+
+const whitePencil = document.querySelector('.profile__avatar-pencil');
+whitePencil.src = whitePencilSrc;
+const logo = document.querySelector('.header__logo');
+logo.src = logoSrc;
+const plus = document.querySelector('.profile__plus-image');
+plus.src = plusSrc;
+// The modal close button always returns NULL and I can't figure out why
+// const whitex = document.querySelector('modal__close-button-white');
+// whitex.src = whiteXSrc;
+
+const closeButtons = document.querySelectorAll(".modal__close-button-logo");
+closeButtons.forEach(button => {
+  button.src = grayXSrc; // Assign the image source
+});
+
+closeDeleteModalButton.addEventListener("click", () => {
```

Comment:
**Could be improved**

You can make a universal handler for any close buttons.

It will look something like this:

```js
// Find all close buttons
const closeButtons = document.querySelectorAll('.modal__close');

closeButtons.forEach((button) => {
  // Find the closest popup only once
  const popup = button.closest('.modal');
  // Set the listener
  button.addEventListener('click', () => closePopup(popup));
});



```

That’s why we make universal css classes for similar elements: here it’s `modal__close` for any close button.

So now you can add 100 modals: their close buttons will be handled automatically.